### PR TITLE
[RTC-349] Add VPS deployment tutorial

### PR DESCRIPTION
**Pull request by blazpie**

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -76,7 +76,8 @@ const sidebars = {
       label: 'Deploying',
       collapsed: true,
       items: [
-        'deploying/fly_io'
+        'deploying/fly_io',
+        'deploying/vps'
```
Comment (roznawsk) [inline sidebars.js:80]: I think you can change the order here - vps is more important so it should be on top.

File: docs/deploying/vps.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
+# Cloud VPS with `cloud-init`
+
+We're encouraging you to use VPS as the most configurable option. In this example, we will show you how to set up Jellyfish media server on [Hetzner Cloud](https://www.hetzner.com/cloud) but you can use these instructions to run Jellyfish on VPS from any provider that supports [cloud-init](https://cloud-init.io/) setup.
```
Comment (roznawsk) [inline docs/deploying/vps.md:3]:
```suggestion
We're encouraging you to use VPS as the most customizable option. In this example, we will show you how to set up Jellyfish media server on [Hetzner Cloud](https://www.hetzner.com/cloud) but you can use these instructions to run Jellyfish on VPS from any provider that supports [cloud-init](https://cloud-init.io/) setup.
```

File: docs/deploying/vps.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+- `SERVER_API_TOKEN=test_token` Server API token is a token you'll need to connect to Jellyfish via API or SDK. Write here a chosen secret value you're going to remember later.
+
+Copy the file and change up those two variables. You're going to need a content of the file in the next section.
+
+### Other tools we'll install with _cloud-init_
+
+As you might noticed we've chosen some packages to be installed:
+
+- `ufw` - Uncomplicated Firewall. We'll need it to allow or block trafic on specific ports [https://help.ubuntu.com/community/UFW](https://help.ubuntu.com/community/UFW).
+- fail2ban - Tool to block traffic recognized as unauthorized [https://www.fail2ban.org/wiki/index.php/Main_Page](https://www.fail2ban.org/wiki/index.php/Main_Page)
```
Comment (roznawsk) [inline docs/deploying/vps.md:109]:
```suggestion
- `fail2ban` - Tool to block traffic recognized as unauthorized [https://www.fail2ban.org/wiki/index.php/Main_Page](https://www.fail2ban.org/wiki/index.php/Main_Page)
```

File: docs/deploying/vps.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+:::caution
+_cloud-init_ will recognize the file as it's configuration only if the file starts with that comment:
+```
+#cloud-config
+...
+```
+So be carefull while copying.
+:::
+
+That's it. You can click 'Create & Buy' an a server will start to configurate.
```
Comment (roznawsk) [inline docs/deploying/vps.md:128]:
```suggestion
That's it. You can click 'Create & Buy' and the server will start to configure.
```

File: docs/deploying/vps.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    - systemctl enable fail2ban
+    - ufw default deny outgoing
+    - ufw default deny incoming
+    - ufw default deny routed
+    - ufw allow in ssh
+    - ufw allow out https
+    - ufw allow out domain
+    - ufw allow in 80/tcp
+    - ufw allow in 50000:50050/udp
+    - ufw allow out from any port 50000:50050 proto udp to any
```
Comment (roznawsk) [inline docs/deploying/vps.md:66]: Perhaphs we should allow more than 50 upd ports in production?

File: docs/deploying/vps.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    - ufw enable
+    - |
+      cat << EOF > /opt/jellyfish/env-file
+      SERVER_API_TOKEN=test_token
+      SECRET_KEY_BASE=$(openssl rand -base64 32)
+      INTEGRATED_TURN_IP=$(ip route get 1.0.0.0 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p')
+      JELLYFISH_ADDRESS=$(ip route get 1.0.0.0 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p')
+      INTEGRATED_TURN_PORT_RANGE=50000-50050
+      CHECK_ORIGIN=false
```
Comment (roznawsk) [inline docs/deploying/vps.md:75]: Does disabling `CHECK_ORIGIN` increase risk of some kind of attack?
If I understand correctly, the reverse proxy makes this flag have no effect.
But there is no reverse proxy in this setup

File: docs/deploying/vps.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+```
+then, in server's terminal you can check current status of cloud init:
+
+```bash
+cloud-init status
+```
+
+Posible responses are `running`, `error` and `done`. The first one informs you that _cloud-init_ is still configuring you server. Second one means that something whent wrong. You can find logs from the _cloud-init_ run in `/var/log/cloud-init-output.log` file. `done` means that your jellyfish server is ready to be used.
+
+
```
Comment (roznawsk) [inline docs/deploying/vps.md:144]: I think you could mention at the end that we can use Jellyfish Dashboard (e.g. the online hosted version) to check if the Jellyfish is working correctly.

**Review by roznawsk — CHANGES_REQUESTED** [review]
Nice tutorial!
It looks like the cloud-init makes the setup process really easy.

Consider passing it through Grammarly to spot any typos.

File: docs/deploying/vps.md (hunk 7 above)
Comment (blazpie) [reply docs/deploying/vps.md:222]: I've added additional paragraph about it

File: docs/deploying/vps.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+## 1. Creating primary IP
+
+We divided the process of creating VPS into two parts (creating IP and configuring server) because before configuring your server with _cloud-init_ using our script, you should have registered your domain to point at the particular IP address.
+
+Let's start with it. In Hetzner's project site go to the section ___Primary IPs___ then click ___Create Primary IP___ button. Choose a data center location for your server and select __IPv4__ protocol. You can also name that IP if you'd like.
+
+## 2. Registering Domain
+
+We'll not suggest you use a specific domain provider, maybe you've got one chosen so we'll describe this step without details that may be different among providers. If you'd like, [you can use Hetzner as well to register a domain](https://www.hetzner.com/domainregistration).
```
Comment (mickel8) [inline docs/deploying/vps.md:15]: Shouldn't we use `to` here?
```suggestion
We'll not suggest you to use a specific domain provider.
Maybe you've got one chosen so we'll describe this step without details that may be different among providers.
If you'd like, [you can use Hetzner as well to register a domain](https://www.hetzner.com/domainregistration).
```

**Review by mickel8 — APPROVED** [review]
Really awesome! :clap: 

Just a couple of non-blocking suggestions so  I am giving the approve

**Review by roznawsk — APPROVED** [review]
Great job!

The tutorial is really easy to follow, and I am glad that we can instantly demonstrate that it works using the Dashboard

360_F_116652906_y9YDrMad6vbHsUkFw9fjVG4uVeFNXcuM (1)

**Pull request merged**